src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -628,11 +628,11 @@ private final class BitmapIndexedMapNode[K, +V](
 
     if ((dataMap & bitpos) != 0) {
       val index = indexFrom(dataMap, mask, bitpos)
-      if (key == getKey(index)) getValue(index) else throw new NoSuchElementException
+      if (key == getKey(index)) getValue(index) else throw new NoSuchElementException("key not found: " + key)
     } else if ((nodeMap & bitpos) != 0) {
       getNode(indexFrom(nodeMap, mask, bitpos)).apply(key, originalHash, keyHash, shift + BitPartitionSize)
     } else {
-      throw new NoSuchElementException
+      throw new NoSuchElementException("key not found: " + key)
```

Comment:
Just curious if there is a style reason not to use `s"key not found: $key"`. They did work to make that not terrible.

Reply:
My usual goto would be `s"key not found: $key"`, but given that [everywhere else](https://github.com/scala/scala/blob/991d2adaa36b74507f968ed38c74f84b7e0ff01b/src/library/scala/collection/immutable/Map.scala#L245) has `"key not found: " + key`, I figured I should try to be consistent with the existing collection code. I'd be happy to ignore the tradition and do the right thing.